src/android/toga_android/widgets/base.py

```diff
@@ -64,4 +84,17 @@ def add_child(self, child):
             child.container = self.container
 
     def rehint(self):
+        # An Android `View` subclass (of which all widgets are subclasses) can
+        # be measured against a measurement specification called a MeasureSpec.
+        # Note that this mutates the widget's `getMeasuredWidth()` and
+        # `getMeasuredHeight()`.
+        self.native.measure(
+            View__MeasureSpec.UNSPECIFIED, View__MeasureSpec.UNSPECIFIED
+        )
+        # We use `at_least()` below because without it, Travertino seems to
+        # decide to use a width of zero.
+        self.interface.intrinsic.width = at_least(self.native.getMeasuredWidth())
```

Comment:
It's also worth noting that on most widgets, there *is* no intrinsic width (or, if there is, it's nothing more than a minimum so that the widget is always visible in both directions. The "atleast on horizontal" sizing is something that is (somewhat ironically) only needed for the three widgets in this example - button, label and text input.

So - the three you've defined here are the exceptions, not the general case. As such, having this definition in `base.rehint()` is probably premature optimization.

Reply:
Okay. I'm cool moving these to the specific classes if you like.

My preference is to leave this as-is, since I haven't implemented any other widgets yet. I picked the approach I did because it was simplest for what I was trying to do in this PR.

Reply:
Yeah - they need to move. It makes total sense in the context of the 4 widgets in play here; it's only when the context of the "next" widget that it's a likely problem.

Reply:
Done!